system/modules/core/classes/Frontend.php

```diff
@@ -469,7 +469,7 @@ protected function jumpToOrReload($intId, $strParams=null, $strForceLang=null)
 			{
 				if ($intId['id'] != $objPage->id  || $blnForceRedirect)
 				{
-					$this->redirect($this->generateFrontendUrl($intId, $strParams, $strForceLang));
```

Comment:
We must forward the language parameter if it's set, or it would potentially alter behavior.